src/components/App/index.js

```diff
@@ -27,12 +28,34 @@ export const App = ({ options }) => {
   const gameOver = new GameOver({
     element: document.querySelector('#gameovermodal'),
     config: config,
-    handleScoreSubmit: (result) => hallOfFame.saveResult(result),
+    handleScoreSubmit: (result) =>
+      hallOfFame.saveResult(result) || hallOfFame.update(),
   });
-  //gameOver.display(); //tymczasowe, jedyna metoda wywołania modala
+
+  //do wykorzystania także, gdy skończą się pytania
+  function onGameOver() {
+    gameOver.display();
+    timer.hide();
+    inGameMode.hidden = true;
+    rankingBtn.hidden = false;
+    playTheGame.hidden = false;
+    btnSettings.hidden = false;
+    switchToHall();
+    gameMode.enableButtons();
+  }
 
   const gameMode = new GameModeSelect(config);
 
+  const timer = new Timer({
+    element: document.querySelector('#progress'),
```

Comment:
musisz to przekazywać w konstruktorze? nie lepiej żeby tylko Timer o tym wiedział?

Reply:
wolę, żeby komponent sam z siebie nie używał document.querySelector, bo to by oznaczało, że komponent "wie", jak jest zbudowany index.html

Reply:
to chyba warto to przegadać na dzisiejszym callu - dla mnie timer jest komponentem uiowym, więc powinien wiedzieć o "swoim" elemencie